src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/ManagedX509ExtensionProcessor.cs

```diff
@@ -221,8 +173,13 @@ public virtual void DecodeX509SubjectKeyIdentifierExtension(byte[] encoded, out
 
         internal static byte[] DecodeX509SubjectKeyIdentifierExtension(byte[] encoded)
         {
-            DerSequenceReader reader = DerSequenceReader.CreateForPayload(encoded);
-            return reader.ReadOctetString();
+            AsnReader reader = new AsnReader(encoded, AsnEncodingRules.DER);
+            ReadOnlyMemory<byte> contents;
+            if (!reader.TryGetPrimitiveOctetStringBytes(out contents))
+            {
+                throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+            }
+            return contents.ToArray();
```

Comment:
reader.ThrowIfNotEmpty() ?

Reply:
Yep, of course. Ironically I just fixed the same error in Pkcs in the other PR for exactly the same structure :-)